### ISSUE TITLE
Adding a new transposed convolution function (similar to `torch.nn.ConvTranspose2d()`)

### ISSUE THREAD

**Issue by andsteing** [google/flax#1872]
Adding a transposed convolution as proposed in
https://github.com/google/jax/pull/5772
would also be very useful when porting models from PyTorch to Flax (as in #1848).

**Comment by codeboy5** [comment]
Hey is this an active issue that is being worked on ?

**Comment by marcvanzee** [comment]
We actually have an implementation already: https://github.com/google/flax/blob/main/flax/linen/linear.py#L447

I don't really understand why we have this issue.

@andsteing can you please clarify?

**Comment by codeboy5** [comment]
According to the docs :- 

`torch.nn.ConvTranspose2d` and `nn.ConvTranspose` are not compatible. `nn.ConvTranspose` is a wrapper around `jax.lax.conv_transpose` which computes a fractionally strided convolution, while `torch.nn.ConvTranspose2d` computes a gradient based transposed convolution. 

@marcvanzee this might be the reason and also why i created this issue.

**Comment by codeboy5** [comment]
Yes, I will be interested to work on it.

**Comment by marcvanzee** [comment]
I just noticed this issue has the "blocked" label. @jheek could you please explain this? I suppose it is blocked on the JAX issue https://github.com/google/jax/pull/5772?

@codeboy5 In that case I guess we have to wait with working on this issue until that one is merged.

**Comment by codeboy5** [comment]
Oh okay. I looked at that issue too, hasn't been any new updates for a year. 
Thanks

**Comment by marcvanzee** [comment]
Hmm I see, maybe you can reply to that issue and ask whether they are planning to merge it soon? Otherwise you could ask them if you can pick up that issue if you are really interested!
